submodules/TelegramCore/Sources/TelegramEngine/UnixTime/UnixTimeFetcher.swift

```diff
@@ -0,0 +1,94 @@
+//
+//  UnixTime.swift
+//  _idx_AccountContext_FFB9E9C3_ios_min9.0
+//
+//  Created by andrew on 10/8/22.
+//
+
+import Foundation
+import SwiftSignalKit
+
+/// API Fetcher, that works with time and timezones
+public protocol UnixTimeFetcher {
+    /// Asynchronously requests current timestamp.
+    ///
+    /// - Returns: Signal with timestamp or error
+    func getTimeStamp() -> Signal<Int32, UnixTimeFetcherError>
+}
+
+/// Error, that could be returned from UnixTimeFetcher
+public enum UnixTimeFetcherError: Error {
+    case illegalUrl
+    case noData
+    case networkError
+    case decodingError
+}
+
+extension TelegramEngine {
+    /// private Implementation of UnixTimeFetcher
+    final class UnixTimeFetcherImpl: UnixTimeFetcher {
```

Comment:
implementation of API fetcher protocol is private, and will not be exposed outside of `TelegramEngine`